Loop/Managers/LoopDataManager.swift

```diff
@@ -61,7 +63,12 @@ final class LoopDataManager {
         overrideHistory: TemporaryScheduleOverrideHistory = UserDefaults.appGroup?.overrideHistory ?? .init(),
         lastPumpEventsReconciliation: Date?,
         analyticsServicesManager: AnalyticsServicesManager,
-        localCacheDuration: TimeInterval = .days(1)
+        localCacheDuration: TimeInterval = .days(1),
+        doseStore: DoseStoreProtocol,
+        glucoseStore: GlucoseStoreProtocol,
+        carbStore: CarbStoreProtocol,
+        cacheStore: PersistenceController = PersistenceController.controllerInAppGroupDirectory(),
```

Comment:
[nit] Not sure if I agree with using a default here. Specifying a default implies (at least to me) that using a different PersistenceController than the one used by the dose, glucose, or carb stores is okay (when it is not). I would remove the default here.

Reply:
Might suggest that cacheStore is removed completely, and settingsStore and DosingDecisionStores are constructed by DDM as well, and passed into LDM.

Reply:
Yep, I removed cacheStore from LDM and constructed settingsStore and dosingdecisionStore in DDM